addons/mrp/models/mrp_production.py

```diff
@@ -674,6 +674,19 @@ def _onchange_workorder_ids(self):
             self._create_workorder()
 
     def write(self, vals):
+        to_confirm = False
+        if 'product_id' in vals and len(self) == 1 and self.state == 'confirmed':
```

Comment:
Why limited only when there one MO in self ?

Reply:
Because I don't think multiple self can happen in this situation, to be safer, I add this. But it can be removed.

Reply:
Well it is true, I understand. But if it is weird to hardcode this limitation. Also with the tree view editable, maybe tomorrow, the product_id won't be readonly anymore.
If you think it is better like that, it is ok for me.